src/libstore/build/derivation-goal.cc

```diff
@@ -392,9 +392,24 @@ Goal::Co DerivationGoal::gaveUpOnSubstitution()
     if (nrFailed != 0) {
         if (!useDerivation)
             throw Error("some dependencies of '%s' are missing", worker.store.printStorePath(drvPath));
+
+        std::string describer;
+        auto firstOutput = drv->outputsAndOptPaths(worker.store).begin()->second.second;
+        if (firstOutput) {
+            describer = worker.store.printStorePath(*firstOutput);
+        } else {
+            describer = drv->name;
+        }
+
         co_return done(BuildResult::DependencyFailed, {}, Error(
-                "%s dependencies of derivation '%s' failed to build",
-                nrFailed, worker.store.printStorePath(drvPath)));
+                (nrFailed == 1
```

Comment:
if there is no `firstOutput`, doesn't this mean the error message will become something unwieldly like "1 dependency of /nix/store/foooooooooooooooooooooooooooobar.drv (derivation '/nix/store/foooooooooooooooooooooooooooobar.drv') failed to build"?

Reply:
No, it uses the derivation name:

```
error: 1 dependency of aggregate-incorrect-hashes (derivation '/nix/store/pcnlw2yyy0lzq03b6a2klhhw1ily9sx0-aggregate-incorrect-hashes.drv') failed to build
```